src/plugins/visualizations/server/migrations/visualization_common_migrations.ts

```diff
@@ -42,3 +45,51 @@ export const commonRemoveDefaultIndexPatternAndTimeFieldFromTSVBModel = (visStat
 
   return visState;
 };
+
+export const commonAddEmptyValueRuleForSavedObjectsWithLessAndGreaterThenZeroRules = (
+  visState: any
+) => {
+  if (visState && visState.type === 'metrics') {
+    const params: any = get(visState, 'params') || {};
+
+    const getRuleWithComparingToZero = (rules: any[] = []) => {
+      const compareWithEqualMethods = ['gte', 'lte'];
+      return last(
+        rules.filter((rule) => compareWithEqualMethods.includes(rule.operator) && rule.value === 0)
```

Comment:
Can you help me with this? Why do we backport for rules gte/gt zero ?
From my understanding the rules that do not work correctly are the rules that are set below zero, right? Except I am missing something here 🤔

Reply:
The fourth step of bug's reproduction was "Apply a color rule (change the background color for metric less than or equal to 0". So, customer expects to see rules, containing 'equal to zero' applied to empty value. As we've decided to treat empty value and zero value differently, we need this backport fo gte zero in addition to lte zero.

Reply:
I was implementing this suggestion: https://github.com/elastic/kibana/issues/100250#issuecomment-843078163

Reply:
Why is it a bug? If the user has series with value greater than 0 and rule about greater than zero it will work. 
cc @alexwizp

Reply:
But where do you see gt?

Reply:
It works only for gte/lte (>= 0, <=0).